kit/src/main/java/com/oracle/javafx/scenebuilder/kit/skeleton/SkeletonWindowController.java

```diff
@@ -76,9 +77,28 @@ private void onCopyAction(ActionEvent event) {
         }
 
         Clipboard.getSystemClipboard().setContent(content);
+    }
+        
+    private String controllerName = null;
+    
+    private SkeletonFileWriter skeletonFileWriter = null;
+    
+    @FXML
+    private void onSaveAction(ActionEvent event) {
+        
+        if (skeletonFileWriter == null) {
+            skeletonFileWriter = new SkeletonFileWriter(()->getStage(), 
+                                                        languageChoiceBox.getSelectionModel()
+                                                                         .getSelectedItem(),
+                                                        textArea.textProperty());
```

Comment:
setting the `languageChoiceBox` selection inside the controller won't change the language extension if a user clicks on "Save As" once and then changes the language and save again.

Reply:
I have reworked it so that it works now. There is also a test for the functionality - but I am not really happy as of now.